[feat] Prepare animated text kit flutter web app for demo

aagarwal1012 opened the pull request:
### Tickets
- Closes #222 

### Description
- Setup firebase hosting for `example` web app.
- Added GitHub actions workflow files for `deployment preview` and `production deployment` to firebase hosting.
- Live `web-app` is accessible at https://animated-text-kit.web.app.
- Sent invitations for the firebase project access to @awhitford and @SirusCodes.

### Reference
> This [tweet](https://twitter.com/luke_pighetti/status/1370333429737390081) has a [gist](https://gist.github.com/lukepighetti/dca8d731aa95adfa86fd01b0c76222d5) that shows how to publish a _flutter web build_ to [Firebase Hosting Preview Channels](https://firebase.google.com/docs/hosting/manage-hosting-resources).

awhitford commented on .github/workflows/deploy-preview.yml:
```diff
@@ -0,0 +1,31 @@
+          java-version: "12.x"
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: stable
```
I think the `stable` channel is the _default_, so you don't really need to be explicit.

:rocket: Advanced would be to offer a _Workflow Input Parameter_ that defaults to `stable`, but can be overridden to be another channel, or a version, and then use that.

aagarwal1012 replied:
Fixed!

awhitford commented on .github/workflows/deploy-preview.yml:
```diff
@@ -0,0 +1,31 @@
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
```
Setting up Java should only be important when you are building for Android -- not targeting Web.

And if it is necessary, why version 12?  (I'm thinking that an LTS version should be used.). My guess is that you copied/pasted this from the `flutter-action` `README`.

aagarwal1012 replied:
Fixed!

awhitford commented on .github/workflows/deploy-preview.yml:
```diff
@@ -0,0 +1,31 @@
+      - uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_ANIMATED_TEXT_KIT }}"
```
💭 `FIREBASE_SERVICE_ACCOUNT_ANIMATED_TEXT_KIT` is long.
Is qualifying a _project secret_ with the project name (`_ANIMATED_TEXT_KIT`) really necessary? 🤔

aagarwal1012 replied:
Actually, I also feel the same but this was the recommended way, but I have updated the secret name to `FIREBASE_SERVICE_ACCOUNT`.

awhitford commented on .github/workflows/deploy-prod.yml:
```diff
@@ -0,0 +1,33 @@
+      - master
+
+jobs:
+  preview:
```
Maybe this job name should be "live" instead of "preview."

aagarwal1012 replied:
Fixed!

awhitford commented on .github/workflows/deploy-prod.yml:
```diff
@@ -0,0 +1,33 @@
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
```
Same issue as above; I don't think setting up Java is necessary.

aagarwal1012 replied:
Fixed!

awhitford commented on example/web/manifest.json:
```diff
@@ -0,0 +1,23 @@
+    "display": "standalone",
+    "background_color": "#0175C2",
+    "theme_color": "#0175C2",
+    "description": "A new Flutter project.",
```
Let's say _Animated Text Kit_ or something.

aagarwal1012 replied:
Fixed.

awhitford commented on example/web/manifest.json:
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "example",
```
Maybe we should say `Animated Text Kit Example`?

aagarwal1012 replied:
Fixed.

awhitford reviewed (changes requested):
This works pretty well and I really like it! 🚀 
* One issue with the preview is that the default text color is black instead of white.  I think this causes Wavy to show nothing (black on black).  This is not necessarily an issue with this commit -- it is likely more of a web issue, but I'm pointing it out in case there is a tweak opportunity, like the `web/manifest.json` file.
* The _Next_ button on the **Colorize** page needs to be clicked twice.  I suspect that there is a stack trace to be investigated.
* Some of the commit activity looks related to Android, rather than Web.

awhitford commented on example/web/index.html:
```diff
@@ -0,0 +1,45 @@
+  <!-- Favicon -->
+  <link rel="icon" type="image/png" href="favicon.png"/>
+
+  <title>example</title>
```
Animated Text Kit Example?

aagarwal1012 replied:
Done👍

awhitford commented on .github/workflows/deploy-preview.yml:
```diff
@@ -0,0 +1,26 @@
+      - uses: actions/checkout@v2
+      - uses: subosito/flutter-action@v1
+      - run: flutter pub get
+      - run: flutter build web --release
```
Minor suggestion:  Style-wise, I like the multi-line approach like:
```yaml
        run: |
          flutter pub get
          flutter build web --release
```

aagarwal1012 replied:
Fixed!

awhitford commented on .github/workflows/deploy-prod.yml:
```diff
@@ -0,0 +1,28 @@
+      - uses: actions/checkout@v2
+      - uses: subosito/flutter-action@v1
+      - run: flutter pub get
+      - run: flutter build web --release
```
Similar style suggestion as above.

aagarwal1012 replied:
Fixed!

awhitford reviewed (changes requested):
One change (`title`), and two minor suggestions.

aagarwal1012 commented:
> One issue with the preview is that the default text color is black instead of white. I think this causes Wavy to show nothing (black on black). This is not necessarily an issue with this commit -- it is likely more of a web issue, but I'm pointing it out in case there is a tweak opportunity, like the web/manifest.json file.  
- Yes, I have fixed this issue, changed the wavy text background-color to `purple`.
</br>

> The Next button on the Colorize page needs to be clicked twice. I suspect that there is a stack trace to be investigated.
- I am not able to debug this issue, it says that `Another exception was thrown: UnimplementedError`.
</br>

> Some of the commit activity looks related to Android, rather than Web.
- When I was setting up the example app for the web, `flutter create .` created some extra files in the `Android` folder which seems to be necessary.
</br>

> Should there be a link to the latest live page on the README?
- Done!
</br>

> Is it possible to embed the version (like in the footer) to be explicit?
- I think this will not be useful because we already have a version badge in the Readme. Generally, app versions should be displayed in the footer, but in the case of `example` app, we don't maintain any version number.

awhitford commented on README.md:
```diff
@@ -2,7 +2,7 @@
 
-<p align="center">A flutter package which contains a collection of some cool and awesome text animations. Recommended package for text animations in Codemagic's Ebook, <a href="https://blog.codemagic.io/flutter-libraries-ebook-by-codemagic/ebook-flutter-libraries-we-love-by-codemagic.pdf">"Flutter libraries we love"</a>.</p><br>
+<p align="center">A flutter package which contains a collection of some cool and awesome text animations. Recommended package for text animations in Codemagic's Ebook, <a href="https://blog.codemagic.io/flutter-libraries-ebook-by-codemagic/ebook-flutter-libraries-we-love-by-codemagic.pdf">"Flutter libraries we love"</a>. Try-out our example app in <a href="https://animated-text-kit.web.app/">live ...
 
```
Minor:  I would say, "Try out our [live example app]."

aagarwal1012 replied:
Fixed!

awhitford reviewed (approved):
I have one minor suggestion for the `README` link, but otherwise this looks great.

aagarwal1012 commented:
Thanks, @awhitford for the thorough review! Now I'm merging this PR.

awhitford commented:
> * The _Next_ button on the **Colorize** page needs to be clicked twice.  I suspect that there is a stack trace to be investigated.

The stack trace looks like:
```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following UnimplementedError was thrown during a scheduler callback:
UnimplementedError

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49  throw_
lib/_engine/engine/canvaskit/layer_scene_builder.dart 196:5                   pushShaderMask
packages/flutter/src/rendering/layer.dart 1779:26                             addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1966:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1131:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1131:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1576:5                              addToScene
packages/flutter/src/rendering/layer.dart 732:5                               buildScene
packages/flutter/src/rendering/view.dart 217:30                               compositeFrame
packages/flutter/src/rendering/binding.dart 457:18                            drawFrame
packages/flutter/src/widgets/binding.dart 874:13                              drawFrame
packages/flutter/src/rendering/binding.dart 319:5                             [_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1144:15                           [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1082:9                            handleDrawFrame
packages/flutter/src/scheduler/binding.dart 998:5                             [_handleDrawFrame]
lib/_engine/engine/platform_dispatcher.dart 896:13                            invoke
lib/_engine/engine/platform_dispatcher.dart 145:5                             invokeOnDrawFrame
lib/_engine/engine.dart 259:45                                                <fn>
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: UnimplementedError
Another exception was thrown: UnimplementedError
Another exception was thrown: UnimplementedError
```

I think this issue is related to https://github.com/flutter/flutter/issues/75495 -- but that appears to be fixed already, and _should_ be part of `stable`.
